src/diffusers/loaders/unet.py

```diff
@@ -364,8 +377,59 @@ def _process_lora(
+            if hotswap:
+                try:
+                    from peft.utils.hotswap import check_hotswap_configs_compatible, hotswap_adapter_from_state_dict
+                except ImportError as exc:
+                    msg = (
+                        "Hotswapping requires PEFT > v0.14. Please upgrade PEFT to a higher version or install it "
+                        "from source."
+                    )
+                    raise ImportError(msg) from exc
+
+            if hotswap:
+
+                def map_state_dict_for_hotswap(sd):
+                    # For hotswapping, we need the adapter name to be present in the state dict keys
+                    new_sd = {}
+                    for k, v in sd.items():
+                        if k.endswith("lora_A.weight") or key.endswith("lora_B.weight"):
+                            k = k[:-7] + f".{adapter_name}.weight"
+                        elif k.endswith("lora_B.bias"):  # lora_bias=True option
+                            k = k[:-5] + f".{adapter_name}.bias"
+                        new_sd[k] = v
+                    return new_sd
+
+            # To handle scenarios where we cannot successfully set state dict. If it's unsucessful,
+            # we should also delete the `peft_config` associated to the `adapter_name`.
+            try:
+                if hotswap:
+                    check_hotswap_configs_compatible(self.peft_config[adapter_name], lora_config)
+                    hotswap_adapter_from_state_dict(
+                        model=self,
+                        state_dict=state_dict,
+                        adapter_name=adapter_name,
+                        config=lora_config,
+                    )
+                    # the hotswap function raises if there are incompatible keys, so if we reach this point we can set
+                    # it to None
+                    incompatible_keys = None
+                else:
+                    inject_adapter_in_model(lora_config, self, adapter_name=adapter_name, **peft_kwargs)
+                    incompatible_keys = set_peft_model_state_dict(self, state_dict, adapter_name, **peft_kwargs)
+            except Exception as e:
+                # In case `inject_adapter_in_model()` was unsuccessful even before injecting the `peft_config`.
+                if hasattr(self, "peft_config"):
+                    for module in self.modules():
+                        if isinstance(module, BaseTunerLayer):
+                            active_adapters = module.active_adapters
+                            for active_adapter in active_adapters:
+                                if adapter_name in active_adapter:
+                                    module.delete_adapter(adapter_name)
+
+                    self.peft_config.pop(adapter_name)
+                logger.error(f"Loading {adapter_name} was unsucessful with the following error: \n{e}")
+                raise
```

Comment:
Can we raise the exception properly instead of logging an error?

Reply:
Also, are we testing if this error is raised?

Reply:
Note that I just copied the pattern from here:

https://github.com/huggingface/diffusers/blob/97abdd2210a540c2e71aee63c80a22723031cd57/src/diffusers/loaders/peft.py#L301-L316

So this is just for consistency.

Reply:
I think we don't need a test that is specific for hotswap failing, as it doesn't really matter why loading the adapter fails. If a test is added, it should probably be something similar to the test that was added when the change was introduced in `peft.py`: https://github.com/huggingface/diffusers/pull/10188/files#diff-b544edcc938e163009735ef4fa963abd0a41615c175552160c9e0f94ceb7f552.

Not sure if it's possible, but maybe that test can be adjusted to trigger this code path?

Reply:
Added a TODO comment.